helper/memory_calculator.go

```diff
@@ -99,26 +99,21 @@ func (m MemoryCalculator) Execute() (map[string]string, error) {
 	if b, err := ioutil.ReadFile(m.MemoryLimitPath); err != nil && !os.IsNotExist(err) {
 		m.Logger.Info("WARNING: Unable to read %s: %s", m.MemoryLimitPath, err)
 	} else if err == nil {
-		s := strings.TrimSpace(string(b))
-		if totalMemory, err = strconv.ParseInt(s, 10, 64); err != nil {
-			return nil, fmt.Errorf("untable to convert memory limit %s to integer\n%w", s, err)
+		if size, err := calc.ParseSize(strings.TrimSpace(string(b))); err != nil {
+			m.Logger.Info("WARNING: Unable to convert memory limit %q from path %q as int: %s", strings.TrimSpace(string(b)), m.MemoryLimitPath, err)
+		} else {
+			totalMemory = size.Value
 		}
 	}
 
 	if totalMemory == UnsetTotalMemory {
 		if b, err := ioutil.ReadFile(m.MemoryInfoPath); err != nil && !os.IsNotExist(err) {
-			m.Logger.Info("WARNING: Unable to read %s: %s", m.MemoryInfoPath, err)
+			m.Logger.Infof(`WARNING: failed to read %q: %s`, m.MemoryInfoPath, err)
 		} else if err == nil {
-			rp := regexp.MustCompile("MemAvailable:\\s*(\\d*)\\skB")
-			m := rp.FindStringSubmatch(string(b))
-
-			if len(m) > 1 {
-				s := m[1]
-				if i, err := strconv.ParseInt(s, 10, 64); err != nil {
-					return nil, fmt.Errorf("untable to convert available memory %s to integer\n%w", s, err)
-				} else {
-					totalMemory = i * calc.Kibi
-				}
+			if mem, err := parseMemInfo(string(b)); err != nil {
+				m.Logger.Infof(`WARNING: failed to parse available memory from path %q: %s`, m.MemoryInfoPath, err)
+			} else {
+				totalMemory = mem
```

Comment:
Should we log a short info "memory calculation is based on x amount of available memory"?
When reading the logs of a container it way be hard to understand why the jvm memory was calculated this way. Especially as it may be impossible to check what the available memory was at startup.